apps/src/lib/node/ledger/ethereum_node/oracle/mod.rs

```diff
@@ -359,14 +398,34 @@ mod test_oracle {
         oracle: Oracle,
         admin_channel: tokio::sync::mpsc::UnboundedSender<TestCmd>,
         eth_recv: tokio::sync::mpsc::Receiver<EthereumEvent>,
+        control_sender: control::Sender,
         blocks_processed_recv: tokio::sync::mpsc::UnboundedReceiver<Uint256>,
         abort_recv: Receiver<()>,
     }
 
+    /// Helper function that starts running the oracle in a new thread, and
+    /// initializes it with a simple default configuration that is appropriate
+    /// for tests.
+    async fn start_with_default_config(
+        oracle: Oracle,
+        control_sender: control::Sender,
+        config: Config,
+    ) -> JoinHandle<()> {
+        let handle = std::thread::spawn(move || {
+            tokio_test::block_on(run_oracle_aux(oracle));
+        });
```

Comment:
https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html

Reply:
you can use something like this https://github.com/anoma/namada/blob/04cf854bafbd343ee2bd93d3436879cbe80ee5b9/apps/src/lib/node/ledger/ethereum_node/oracle.rs#L120-L142

Reply:
at least until we get rid of `web30`